core/chains/evm/logpoller/log_poller_internal_test.go

```diff
@@ -0,0 +1,316 @@
+package logpoller
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind/backends"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/rawdb"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/jackc/pgconn"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap/zapcore"
+	"go.uber.org/zap/zaptest/observer"
+
+	"github.com/smartcontractkit/chainlink/core/chains/evm/client"
+	"github.com/smartcontractkit/chainlink/core/gethwrappers/generated/log_emitter"
+	"github.com/smartcontractkit/chainlink/core/internal/testutils"
+	"github.com/smartcontractkit/chainlink/core/internal/testutils/pgtest"
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/utils"
+)
+
+var (
+	EmitterABI, _ = abi.JSON(strings.NewReader(log_emitter.LogEmitterABI))
+)
+
+// Validate that filters stored in log_filters_table match the filters stored in memory
+func validateFiltersTable(t *testing.T, lp *logPoller, orm *ORM) {
+	filters, err := orm.LoadFilters()
+	require.NoError(t, err)
+	require.Equal(t, len(filters), len(lp.filters))
+	for name, dbFilter := range filters {
+		dbFilter := dbFilter
+		memFilter, ok := lp.filters[name]
+		require.True(t, ok)
+		assert.True(t, memFilter.Contains(&dbFilter),
+			fmt.Sprintf("in-memory Filter %s is missing some addresses or events from db Filter table", name))
+		assert.True(t, dbFilter.Contains(&memFilter),
+			fmt.Sprintf("db Filter table %s is missing some addresses or events from in-memory Filter", name))
+	}
+}
+
+func TestLogPoller_RegisterFilter(t *testing.T) {
```

Comment:
@reductionista I think this test can be broken down into a blackbox ORM test which cover the ORM methods (insert/delete/load) and then a blackbox logpoller one which uses the now exposed Filter to verify the cache invalidation stuff